### PR TITLE
RATIS-1908. Keep configurations doc updated

### PR DESCRIPTION
**Pull request by SzyWilliam**
See https://issues.apache.org/jira/browse/RATIS-1908

### CODE REVIEW

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM~
I forgot to update the documentation when I introduced the `raft.server.snapshot.trigger-when-stop.enabled` parameter, I will take note of that later. Thanks William for adding docs!

File: ratis-docs/src/site/markdown/configurations.md [hunk 1]
```diff
@@ -199,6 +199,16 @@ treat the peer as caught-up. Increase this number when write throughput is high.
 | **Type**        | double, ranging from (0.0,1.0)                |
 | **Default**     | 0.9                                           |
 
+### Read After Write - Configurations related to read-after-write-consistency
+
+
+| **Property**    | `raft.server.read.read-after-write-consistent.write-index-cache.expiry-time`   |
```
Comment (szetszwo) [inline ratis-docs/src/site/markdown/configurations.md:205]: >     ******* interface org.apache.ratis.server.RaftServerConfigKeys$Read$ReadAfterWriteConsistent *******
>     constant: PREFIX = raft.server.read-after-write-consistent
>     key: raft.server.read-after-write-consistentwrite-index-cache.expiry-time (org.apache.ratis.util.TimeDuration, default=60s)

Above is the output of `RaftServerConfigKeys.main()`.  We should fix the conf key as below.

```java
+++ b/ratis-server-api/src/main/java/org/apache/ratis/server/RaftServerConfigKeys.java
@@ -215,9 +215,9 @@ public interface RaftServerConfigKeys {
     }
 
     interface ReadAfterWriteConsistent {
-      String PREFIX = RaftServerConfigKeys.PREFIX + ".read-after-write-consistent";
+      String PREFIX = Read.PREFIX + ".read-after-write-consistent";
 
-      String WRITE_INDEX_CACHE_EXPIRY_TIME_KEY = PREFIX + "write-index-cache.expiry-time";
+      String WRITE_INDEX_CACHE_EXPIRY_TIME_KEY = PREFIX + ".write-index-cache.expiry-time";
       /** Must be larger than {@link Read#TIMEOUT_DEFAULT}. */
       TimeDuration WRITE_INDEX_CACHE_EXPIRY_TIME_DEFAULT = TimeDuration.valueOf(60, TimeUnit.SECONDS);
```

**Review by szetszwo — COMMENTED** [review]
@SzyWilliam , thanks for working on this!  Please see the comment inlined.

File: ratis-docs/src/site/markdown/configurations.md (hunk 1 above)
Comment (SzyWilliam) [reply ratis-docs/src/site/markdown/configurations.md:205]: Thanks for pointing out! Let me fix these in this patch.

**Comment by SzyWilliam** [comment]
@szetszwo, thanks for the reviews! Made changes on code, PTAL.

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Pull request merged**